src/pybind/mgr/cephadm/module.py

```diff
@@ -2141,6 +2148,22 @@ def _apply_service_spec(self, spec: ServiceSpec) -> str:
             raise OrchestratorError('cannot scale %s service below 1' % (
                 spec.service_type))
 
+        if spec.placement.count is not None:
+            host_count = len(self.inventory.keys())
+            max_count = self.max_count_per_host
+            if spec.service_type in ['mon', 'mgr']:
+                if spec.placement.count > max(5, host_count):
+                    raise OrchestratorError(
+                        (f'The maximum number of {spec.service_type} daemons allowed with {host_count} hosts is {max(5, host_count)}.'))
+            elif spec.service_type != 'osd':
+                if spec.placement.count > (max_count * host_count):
```

Comment:
```
2021-04-05T16:59:52.733 INFO:journalctl@ceph.mon.gibba033.gibba033.stdout:Apr 05 16:59:52 gibba033 conmon[30784]: audit 2021-04-05T16:59:51.218161+0000 mgr.gibba029.qchlxw (mgr.14168) 218 : audit [DBG] from='client.14430 -' entity='client.admin' cmd=[{"prefix": "orch devi
ce ls", "target": ["mon-mgr", ""]}]: dispatch
2021-04-05T16:59:53.410 INFO:teuthology.orchestra.run.gibba029.stderr:Error EINVAL: Traceback (most recent call last):
2021-04-05T16:59:53.411 INFO:teuthology.orchestra.run.gibba029.stderr:  File "/usr/share/ceph/mgr/mgr_module.py", line 1335, in _handle_command
2021-04-05T16:59:53.411 INFO:teuthology.orchestra.run.gibba029.stderr:    return self.handle_command(inbuf, cmd)
2021-04-05T16:59:53.412 INFO:teuthology.orchestra.run.gibba029.stderr:  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 167, in handle_command
2021-04-05T16:59:53.412 INFO:teuthology.orchestra.run.gibba029.stderr:    return dispatch[cmd['prefix']].call(self, cmd, inbuf)
2021-04-05T16:59:53.412 INFO:teuthology.orchestra.run.gibba029.stderr:  File "/usr/share/ceph/mgr/mgr_module.py", line 389, in call
2021-04-05T16:59:53.413 INFO:teuthology.orchestra.run.gibba029.stderr:    return self.func(mgr, **kwargs)
2021-04-05T16:59:53.413 INFO:teuthology.orchestra.run.gibba029.stderr:  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 107, in <lambda>
2021-04-05T16:59:53.414 INFO:teuthology.orchestra.run.gibba029.stderr:    wrapper_copy = lambda *l_args, **l_kwargs: wrapper(*l_args, **l_kwargs)  # noqa: E731
2021-04-05T16:59:53.414 INFO:teuthology.orchestra.run.gibba029.stderr:  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 96, in wrapper
2021-04-05T16:59:53.415 INFO:teuthology.orchestra.run.gibba029.stderr:    return func(*args, **kwargs)
2021-04-05T16:59:53.415 INFO:teuthology.orchestra.run.gibba029.stderr:  File "/usr/share/ceph/mgr/orchestrator/module.py", line 1092, in _apply_rgw
2021-04-05T16:59:53.416 INFO:teuthology.orchestra.run.gibba029.stderr:    return self._apply_misc([spec], dry_run, format, no_overwrite)
2021-04-05T16:59:53.416 INFO:teuthology.orchestra.run.gibba029.stderr:  File "/usr/share/ceph/mgr/orchestrator/module.py", line 1031, in _apply_misc
2021-04-05T16:59:53.416 INFO:teuthology.orchestra.run.gibba029.stderr:    raise_if_exception(completion)
2021-04-05T16:59:53.417 INFO:teuthology.orchestra.run.gibba029.stderr:  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 224, in raise_if_exception
2021-04-05T16:59:53.417 INFO:teuthology.orchestra.run.gibba029.stderr:    raise e
2021-04-05T16:59:53.418 INFO:teuthology.orchestra.run.gibba029.stderr:UnboundLocalError: local variable 'max_count' referenced before assignment
2021-04-05T16:59:53.418 INFO:teuthology.orchestra.run.gibba029.stderr:
2021-04-05T16:59:53.984 DEBUG:teuthology.orchestra.run:got remote process result: 22
```

https://pulpito.ceph.com/kchai-2021-04-05_10:46:36-rados-wip-kefu-testing-2021-04-05-1650-distro-basic-gibba/6021979/

Reply:
Think this was caused by me only defining max_count and host_count inside one of the if blocks. Moved it outside so it should always be defined now and this shouldn't happen anymore.